pkg/cmd/cli/cmd/version.go

```diff
@@ -0,0 +1,163 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	etcdversion "github.com/coreos/etcd/version"
+
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kubeversion "k8s.io/kubernetes/pkg/version"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	"github.com/openshift/origin/pkg/version"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	versionLong = `
+Display client and server versions.`
+)
+
+type VersionOptions struct {
+	BaseName string
+	Out      io.Writer
```

Comment:
store the clients and that clientconfig (we should fix that remote call someday) here.